tests/suite/upstream_settings_test.go

```diff
@@ -0,0 +1,414 @@
+	})
+
+	AfterAll(func() {
+		Expect(resourceManager.DeleteNamespace(namespace)).To(Succeed())
+	})
+
+	When("UpstreamSettingsPolicy are applied to the resources", func() {
+		snippetsFilter := []string{
+			"upstream-settings-policy/valid-usps.yaml",
+		}
+
+		BeforeAll(func() {
+			Expect(resourceManager.ApplyFromFiles(snippetsFilter, namespace)).To(Succeed())
+		})
+
+		AfterAll(func() {
+			Expect(resourceManager.DeleteFromFiles(snippetsFilter, namespace)).To(Succeed())
+		})
+
+		Specify("usPolicies are accepted", func() {
+			usPolicies := []string{
+				"coffee-svc-usp",
+				"grpc-svc-usp",
+				"tea-multiple-svc-usp",
+			}
+
+			for _, name := range usPolicies {
+				nsname := types.NamespacedName{Name: name, Namespace: namespace}
+
+				err := waitForUSPolicyStatus(nsname, metav1.ConditionTrue, v1alpha2.PolicyReasonAccepted)
+				Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("%s was not accepted", name))
+			}
+		})
+
+		Context("verify working traffic", func() {
+			It("should return a 200 response for HTTPRoute", func() {
+				port := 80
+				if portFwdPort != 0 {
+					port = portFwdPort
+				}
+				baseCoffeeURL := fmt.Sprintf("http://cafe.example.com:%d%s", port, "/coffee")
+				baseTeaURL := fmt.Sprintf("http://cafe.example.com:%d%s", port, "/tea")
+
+				Eventually(
+					func() error {
+						return expectRequestToSucceed(baseCoffeeURL, address, "URI: /coffee")
+					}).
+					WithTimeout(timeoutConfig.RequestTimeout).
+					WithPolling(500 * time.Millisecond).
+					Should(Succeed())
+
+				Eventually(
+					func() error {
+						return expectRequestToSucceed(baseTeaURL, address, "URI: /tea")
+					}).
+					WithTimeout(timeoutConfig.RequestTimeout).
+					WithPolling(500 * time.Millisecond).
+					Should(Succeed())
+			})
+		})
+
+		Context("nginx directives", func() {
+			var conf *framework.Payload
+
+			BeforeAll(func() {
+				podNames, err := framework.GetReadyNGFPodNames(k8sClient, ngfNamespace, releaseName, timeoutConfig.GetTimeout)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(podNames).To(HaveLen(1))
+
+				ngfPodName := podNames[0]
+
+				conf, err = resourceManager.GetNginxConfig(ngfPodName, ngfNamespace)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			// TODO: important
+			// The directive file and field value need to be updated based on the
+			// implementation of the UpstreamSettingsPolicy and how they are specified in the config files.
+			DescribeTable("are set properly for",
+				func(expCfgs []framework.ExpectedNginxField) {
+					for _, expCfg := range expCfgs {
+						Expect(framework.ValidateNginxFieldExists(conf, expCfg)).To(Succeed())
+					}
+				},
+				Entry("HTTPRoute", []framework.ExpectedNginxField{
+					{
+						Directive:             "zone default_coffee_80",
+						Value:                 "128k",
+						Upstreams:             []string{"default_coffee_80"},
+						File:                  "nginx.conf",
+						ValueSubstringAllowed: true,
```

Comment:
because `zone_size` has two variables after it --> upstream name and zone_size value? Am I understanding this field wrong that is looks for value in other substrings of the result?

Reply:
Hmm the directive you are searching for contains the upstream name, though. Will that work? Maybe @sjberman can help out here.